src/backend/distributed/utils/background_jobs.c

```diff
@@ -336,12 +567,21 @@ CitusBackgroundTaskQueueMonitorMain(Datum arg)
 	/* connect to database, after that we can actually access catalogs */
 	BackgroundWorkerInitializeConnectionByOid(databaseOid, extensionOwner, 0);
 
+	/*
+	 * save old context until monitor loop exits, we use backgroundTaskContext for
+	 * all allocations.
+	 */
+	MemoryContext firstContext = CurrentMemoryContext;
+	MemoryContext backgroundTaskContext = AllocSetContextCreate(TopMemoryContext,
+																"BackgroundTaskContext",
+																ALLOCSET_DEFAULT_SIZES);
+
 	StartTransactionCommand();
 	PushActiveSnapshot(GetTransactionSnapshot());
 
-	/* load database name and copy to a memory context that survives the transaction */
+	/* make databasename alive during queue monitor lifetime */
+	MemoryContext oldContext = MemoryContextSwitchTo(backgroundTaskContext);
```

Comment:
Why did you move this up? It seems like that's unnecessary, any allocations that `get_database_name` would now also be part of the backgroundMemoryContext. But those shouldn't be necessary because we re-allocate databasename with `pstrdup`